workbench-core/datasets/src/dataSetService.ts

```diff
@@ -295,15 +313,15 @@ export class DataSetService {
 
       if (
         !targetDS.externalEndpoints ||
-        !_.find(targetDS.externalEndpoints, (ep) => ep === targetEndpoint.name)
+        !_.find(targetDS.externalEndpoints, (ep) => ep === targetEndpoint.id)
```

Comment:
this should be name, the name is being stored in ddb rather than the id

Reply:
It looks like I missed a few locations, and now have some concerns regarding the change. Either where I made the change or elsewhere where `id` is used should have failed a test somewhere.

Reply:
I think we should complete the change I made in an earlier PR and swap `externalEndpoint.id` for `externalEndpoint.name` everywhere since `name` is also unique and accessible in `addExternalEndpointForUser/Group` where `is` is not.

Reply:
what do you mean by 'accessible'?
>  accessible in addExternalEndpointForUser/Group where id is not.

Reply:
in `addExternalEndpointForUser/Group` there is a check to see if the endpoint already exists for the dataset. We cant check `dataset.externalEndpoint` for an endpoint id because the endpoint doesnt have one yet. We can check the `externalEndpoint` list for an endpoint name, because that is passed in as a parameter.

Reply:
either way, if you think this is important, please open a ticket. It really is out of scope for this PR. This PR just made it consistent enough to work again. The ticket should also investigate why the unit tests passed.